src/world.c

```diff
@@ -1427,6 +1426,138 @@ void FixSayTeamToSpecs()
 
 int skip_fixrules = 0;
 
+// This function determines the actual game mode, include game type, and sub-mode,
+// which is part of the serverinfo reply.
+// Format is: mode-submode[-submode]
+void SetMode(void)
+{
+	char mode[64] = {};
+
+	switch (current_umode)
```

Comment:
Better add UserModeToString(mode) instead of this switch() which essentially such function, so it could be re-used, wonder if we already have such function?

Reply:
Hmm, I am not sure if I get the idea behind adding another function.
We need a switch anyhow. Here, or in the new UserModeToString() function, doesn't matter that much, does it?

Or maybe I miss your idea!
Maybe add a pseudo code?

Reply:
As I wrote, "so it could be re-used", thats the point. So later on when you would need convert user mode from integer to string you would have such funciton already. 

So I had to check actual code and we already have [um_list](https://github.com/QW-Group/ktx/blob/087a8a57256b737c6c903d12534e360c11150bd9/src/commands.c#L4277) array of structs, it could be used for mapping user mode integer to string (actually it already used in some places). Also, we have `um_idx_byname()` which do opposite to what you need but it should give you a hint why you need such things.

There is some issues you probably would have to figure out yourself, like names in `um_list` use **`-`** (hyphen) and I suggested you to separate values in `serverinfo mode` with **`-`** (hyphen) so probably you would have to switch to **`:`** (colon), I think it actually would be the right choice (like PATH environment variable do in unix and such).

Reply:
Please take a look on the update.
I came up with this:
`strlcat(mode, (char *)um_name_byidx(current_umode-1), sizeof(mode));`